src/Twitter.php

```diff
@@ -1435,15 +1452,106 @@ protected function performPost(string $method, $data, Http\Client $client) : Htt
      *
      * @param int|string $id
      * @param array $params
+     * @throws Exception\InvalidArgumentException if the value is neither an integer nor a string
      */
     protected function createUserParameter($id, array $params) : array
     {
-        if ($this->validInteger($id)) {
+        if (! is_string($id) && ! is_int($id)) {
+            throw new Exception\InvalidArgumentException(sprintf(
+                '$id must be an integer or a string, received %s',
+                is_object($id) ? get_class($id) : gettype($id)
+            ));
+        }
+
+        if (0 !== $this->validInteger($id)) {
             $params['user_id'] = $id;
             return $params;
         }
 
+        if (! is_string($id)) {
+            throw new Exception\InvalidArgumentException(sprintf(
+                '$id must be an integer or a string, received %s',
+                gettype($id)
+            ));
+        }
+
         $params['screen_name'] = $this->validateScreenName($id);
         return $params;
     }
+
+    /**
+     * Prepares a list of identifiers for use with endpoints accepting lists of users.
+     *
+     * The various `lookup` endpoints allow passing either:
+     *
+     * - a single user identifier
+     * - a single screen name
+     * - a list of user identifiers
+     * - a list of screen names
+     *
+     * This method checks for each of these conditions. For scalar $ids, it
+     * proxies to {@link createUserParameter}. Otherwise, it checks to ensure
+     * that all values are of the same type. For identifiers, it then
+     * concatenates the values and returns them in the `user_id` parameter.
+     * For screen names, it validates them first, before concatenating and
+     * returning them via the `screen_name` parameter.
+     *
+     * @param int|string|array $ids
+     * @throws Exception\InvalidArgumentException for a non-int/string/array $ids value.
+     * @throws Exception\InvalidArgumentException if an array of $ids exceeds 100 items.
+     * @throws Exception\InvalidArgumentException if an array of $ids are not all of the same type.
+     * @throws Exception\InvalidArgumentException if any screen name element is invalid.
+     */
```

Comment:
They look ok to me. Once they are implemented (and the other PR as well, I'll plug it into my projects and see if anything breaks.  :)

Cheers!
=C=